src/psyclone/psyir/symbols/symbol_table.py

```diff
@@ -1396,6 +1396,24 @@ def resolve_imports(self, container_symbols=None, symbol_target=None):
                                             symbol, symbol_target):
                     continue
 
+                # Determine if there is an Unresolved Symbol in a
+                # descendent symbol table that matches the name of the
+                # symbol we are importing and if so, move it to this
+                # symbol table if a symbol with the same name does not
+                # already exist in this symbol table.
+
+                # Import here to avoid circular dependencies
+                # pylint: disable=import-outside-toplevel
+                from psyclone.psyir.nodes import ScopingNode
+                for scoping_node in self.node.walk(ScopingNode):
+                    symbol_table = scoping_node.symbol_table
+                    if symbol.name in symbol_table:
+                        test_symbol = symbol_table.lookup(symbol.name)
+                        if (type(test_symbol) == Symbol and test_symbol.is_unresolved
+                                and not test_symbol.name in self):
+                            symbol_table.remove(test_symbol)
+                            self.add(test_symbol)
```

Comment:
This is probably ok. There is the small worry that there is still one "continue # It doesn't come from this import" condition below and we would leave the tree altered anyway, but maybe we can leave with this until this represents a problem.

Reply:
This test is within `if symbol.name in self` so will only be checked if the symbol is in the container so the final modified state is probably what is expected at this point? What I worry about is if the symbol should belong where it was originally defined, as there is a use statement at that level, but perhaps I am just saying what you are in a different way. I will hopefully add appropriate tests!

Reply:
I've added a test that shows the problem if we try to inline and there is an existing wildcard use statement in the routine that would resolve the symbol.